pkgs/games/prismlauncher/wrapper.nix

```diff
@@ -21,71 +24,83 @@
 , msaClientID ? null
 , gamemodeSupport ? stdenv.isLinux
 , textToSpeechSupport ? stdenv.isLinux
+, waylandSupport ? false
+, shellWrapper ? waylandSupport
 , jdks ? [ jdk17 jdk8 ]
 , additionalLibs ? [ ]
 , additionalPrograms ? [ ]
 }:
+
+assert lib.assertMsg (waylandSupport -> stdenv.isLinux) "waylandSupport is only available on Linux";
+assert lib.assertMsg (waylandSupport -> shellWrapper) "waylandSupport requires shellWrapper";
+
 let
-  prismlauncherFinal = prismlauncher-unwrapped.override {
+  super = prismlauncher-unwrapped.override {
     inherit msaClientID gamemodeSupport;
   };
+
+  runtimePrograms = [
+    # xorg.xrandr needed for LWJGL [2.9.2, 3) https://github.com/LWJGL/lwjgl/issues/128
+    xorg.xrandr
+    mesa-demos # need glxinfo
+  ]
+  ++ additionalPrograms;
+
+  runtimeLibs = [
+    # xorg
+    xorg.libX11
+    xorg.libXext
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXxf86vm
+    # lwjgl
+    libpulseaudio
+    libGL
+    glfw
+    openal
+    stdenv.cc.cc.lib
+    # oshi
+    udev
+  ]
+  ++ lib.optional gamemodeSupport gamemode.lib
+  ++ lib.optional textToSpeechSupport flite
+  ++ additionalLibs;
```

Comment:
these could be moved back to their original spot. it makes more sense for them to be scoped to `qtWrapperArgs` as that's the only place they're used - plus it results in slightly less thunks :p

Reply:
If you say so... I really hate deeply nested code. I was mildly offended when I first read the original, using `with` unnecessarily, extra concatenation, having deeply-nested `let` blocks, making the reading of `symlinkJoin`'s attrs arg unpleasant. Flat code and simple lists are for sure the way to go. Can you concede on this?

Reply:
The original code wasn't really deeply nested? The reason for the separated "unnecessary" `with` block was to improve static analysis of the code. I guess it's arguable if `with` was needed at all.

Reply:
i wouldn't call what's currently in master "deeply nested" honestly; there's only one `let; in` statement in the entire main body of the expression. the use of `with xorg;` is also not *super* pointless imo, but i can a see it being a very small qol thing.

but overall, i still think this is a very subjective cosmetic change and i don't see a reason to adopt it, especially when evaluation time is a bit better off without it. it's fine how it is in master really, but i would welcome the removal of `with xorg;` and the extra concatenation here